### PR TITLE
Generate and expose txID

### PR DESCRIPTION
**Pull request by Developerayo**
```bash
console.log
      Created Fractional token:  {
        "asset_id": "027a9f437e2ce037d79a1d09bbf7c9e8985f1af2cabd7bebb51b4195a636767bb1",
        "asset_balance": "1",
        "nft_id": "015722b2498fb1a4670c075e7ef77bfd563c80272d9b89474a0e20c81417970534"
      }

      at Object.<anonymous> (tests/actions.test.ts:261:17)

    console.log
      Fractional asset info retrieved

      at Object.<anonymous> (tests/actions.test.ts:280:17)

    console.log
      Transaction ID: c96dbb71acf1822ddf0719de9316d7f222c81c2b15f7a1f24a8ff934f465aa05

      at Object.<anonymous> (tests/actions.test.ts:302:15)

    console.log
      Created dataset:  {
        "dataset_address": "027a9f437e2ce037d79a1d09bbf7c9e8985f1af2cabd7bebb51b4195a636767bb1",
        "dataset_parent_nft_address": "015722b2498fb1a4670c075e7ef77bfd563c80272d9b89474a0e20c81417970534"
      }

      at Object.<anonymous> (tests/actions.test.ts:304:15)

    console.log
      Transaction ID: 2a703facb68f3b49715845aa398b4c8aaa3d4db4d09161b06d7f80991d93820b

      at Object.<anonymous> (tests/actions.test.ts:326:15)

    console.log
      Initiated dataset contribution:  {
        "dataset_contribution_id": "iCR9i9SSYJTWfzLxQV9fPRTKUHASywHAg2XdpcxxBkfQ4GX2D",
        "collateral_asset_address": "00cf77495ce1bdbf11e5e45463fad5a862cb6cc0a20e00e658c4ac3355dcdc64bb",
        "collateral_amount_taken": "1000000000"
      }

      at Object.<anonymous> (tests/actions.test.ts:328:15)

    console.log
      Transaction ID: e8a2a8d3548d35e2c2c22b95b7543d624f560e746533d13ec344780f8bfcb7df

      at Object.<anonymous> (tests/actions.test.ts:349:15)

    console.log
      Completed dataset contribution:  {
        "collateral_asset_address": "00cf77495ce1bdbf11e5e45463fad5a862cb6cc0a20e00e658c4ac3355dcdc64bb",
        "collateral_amount_refunded": "1000000000",
        "dataset_child_nft_address": "025d8ade1dc932bdcdf9a2e0f4267761a4952a9a658cd88c48177f1278be2f93fb",
        "to": "00c4cb545f748a28770042f893784ce85b107389004d6a0e0d6d7518eeae1292d9",
        "data_location": "ipfs://QmTest...",
        "data_identifier": "test-data-001"
      }

      at Object.<anonymous> (tests/actions.test.ts:351:15)

    console.log
      Dataset info retrieved

      at Object.<anonymous> (tests/actions.test.ts:364:15)

    console.log
      Transaction ID: 3ee0a9f24a2b5780eae87ea82e6e5eddc872e75f6771e5dccd8bf1a957dcd47b

      at Object.<anonymous> (tests/actions.test.ts:380:15)
```

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -51,6 +53,30 @@ export class NuklaiCoreApiClient {
   }
 }
 
+export interface ActionResult {
```
Comment (kpachhai) [inline src/client.ts:56]: I'm a little confused about this. This wouldn't apply to all the results though for different actions though

File: src/client.ts [hunk 2]
```diff
@@ -51,6 +53,30 @@ export class NuklaiCoreApiClient {
...
+  asset_id: string;
+  dataset_address?: string;
+  dataset_contribution_id?: string;
+  marketplace_asset_address?: string;
+}
+
+export interface TransactionResult {
```
Comment (kpachhai) [inline src/client.ts:63]: I think as part of this, we should also return ActionInput where we return the parameters that went to execute each action. Also add the sponsor to this struct so we know who executed the action

File: src/client.ts (hunk 1 above)
Comment (Developerayo) [reply src/client.ts:91]: Right! Extended it

File: src/client.ts (hunk 2 above)
Comment (Developerayo) [reply src/client.ts:109]: 💯

**Review by kpachhai — APPROVED** [review]

**Pull request merged**